File: proj4a/Ackermann/ackermann.c
#include <stdio.h>
#include <time.h>

int ackermann(int m, int n, int p) {
    if (p == 0) {
        return m + n;
    } else if (n == 0) {
        if (p == 1) {
            return 0;
        } else if (p == 2) {
            return 1;
        } else {
            return m;
        }
    } else {
        return ackermann(m, ackermann(m, n - 1, p), p - 1);
    }
}

int main() {
    clock_t start = clock();
    int a = ackermann(20,5,2);
    clock_t end = clock();
    printf("Resultado = %d\n", a);
    double elapsed = ((double)(end - start)) / CLOCKS_PER_SEC;
    printf("Tempo de execução: %.3f segundos\n", elapsed);
    return 0;
}

File: proj4a/Ackermann/Ackermann.java
import java.time.Duration;
import java.time.Instant;

public class Ackermann {

    public static int ackermann(int m, int n, int p) {
        if (p == 0) {
            return m + n;
        } else if (n == 0) {
            if (p == 1) {
                return 0;
            } else if (p == 2) {
                return 1;
            } else {
                return m;
            }
        } else {
            return ackermann(m, ackermann(m, n - 1, p), p - 1);
        }
    }

    public static void main(String[] args) {
        Instant start = Instant.now();
        int a = ackermann(10, 5, 2);
        Instant end = Instant.now();
        System.out.println("Resultado = " + a);
        double elapsed = Duration.between(start, end).toMillis() / 1000.0;
        System.out.printf("Tempo de execução: %.3f segundos%n", elapsed);
    }
}
